app/quest-boost/claim/[boostId]/page.tsx

```diff
@@ -0,0 +1,207 @@
+"use client";
+
+import React, { useCallback, useEffect, useState } from "react";
+import styles from "@styles/questboost.module.css";
+import { getBoostById, getQuestBoostClaimParams } from "@services/apiService";
+import { useAccount } from "@starknet-react/core";
+import Button from "@components/UI/button";
+import { useNotificationManager } from "@hooks/useNotificationManager";
+import { NotificationType, TransactionType } from "@constants/notifications";
+import { CDNImage } from "@components/cdn/image";
+import Lottie from "lottie-react";
+import claimWinnerLottie from "@public/visuals/sq_claim.json";
+import claimLoserLottie from "@public/visuals/sq_looser.json";
+import { hexToDecimal } from "@utils/feltService";
+import { boostClaimCall } from "@utils/callData";
+import BoostClaimStatusManager from "@utils/boostClaimStatusManager";
+import { TOKEN_ADDRESS_MAP } from "@utils/constants";
+
+type BoostQuestPageProps = {
+  params: {
+    boostId: string;
+  };
+};
+
+export default function Page({ params }: BoostQuestPageProps) {
+  const { address, account } = useAccount();
+  const { boostId } = params;
+  const [boost, setBoost] = useState<Boost>();
+  const [sign, setSign] = useState<Signature>(["", ""]);
+  const { addTransaction } = useNotificationManager();
+  const [displayCard, setDisplayCard] = useState<boolean>(false);
+  const [displayLottie, setDisplayLottie] = useState<boolean>(true);
+  const [transactionHash, setTransactionHash] = useState<string>("");
+
+  const fetchPageData = async () => {
+    const boostInfo = await getBoostById(boostId);
+    setBoost(boostInfo);
+  };
+
+  useEffect(() => {
+    fetchPageData();
+  }, []);
+
+  const fetchBoostClaimParams = async (): Promise<Signature> => {
+    let formattedSign: Signature = ["", ""];
+    try {
+      if (!boost?.id || !address) return formattedSign;
+      const res = await getQuestBoostClaimParams(boost.id);
+      formattedSign = [res?.r, res?.s];
+      return formattedSign;
+    } catch (err) {
+      console.log(err);
+      return formattedSign;
+    }
+  };
+
+  const handleClaimClick = async () => {
+    const signature = await fetchBoostClaimParams();
+    if (!signature) return;
+    setSign(signature);
+  };
+
+  const getTokenName = useCallback(() => {
```

Comment:
maybe worth putting the whole thing in a service considering your reusing on both pages

Reply:
done